### DISCUSSION TITLE
Let's talk ticketing/intake systems. How is your OSPO handling requests for contribution, license compliance, etc.?

### DISCUSSION THREAD

**Discussion by annania** [todogroup/ospology#139] [OSPO Discussions]
OSPOs often operate as a center of excellence within an organization and consult with Engineering, Product, Legal, etc. Would love to hear how OSPOs are managing these requests. Are you using a commercial product, open source, or did you make your own homegrown solution?

**Comment by draco2003** [comment]
A little context about our workflow / policies and then the details. We have engineers use their personal GitHub Handles for all public facing contributions, but we require work related contributions to be forked under our GitHub Contribs Organization. Based on our internal processes the two most frequent activities we handle for helping our Engineers engage in open source projects are:

### Adding Users to our Github Orgs

 This step is handled by a homegrown solution that allows engineers to submit a PR to a mapping file which contains their GitHub handle and Internal Username. Then on Merge update the GitHub Org Memberships accordingly. Same system manages converting departed engineers to outside collaborators. 

### Forking Repositories to Contribute to

For this we open sourced a GitHub action that allows a PR based workflow in order to self-service fork a repository into our [Contribs Org](https://github.com/wayfair-contribs) . It takes an allow list of Licenses to check against before forking. ( https://github.com/marketplace/actions/github-forker )

**Reply by annania** [reply]
Thank you for your response! I know companies are also looking to solve how to manage tracking of github memberships and correlate with internal employee ids. So is the contribution process fully automated/self-service with this or is there a review board gate-keeper that needs to review CLAs and approve contributions?

**Reply by draco2003** [reply]
We're still finalizing the roll out so it's not fully automated/self-service currently, but once any kinks are worked out the goal is that it will be fully automated/self-service at least for these two activities. 
Example for joining our Github Orgs, the PR that an engineer submits requires their manager as a reviewer which qualifies as "sign off" prior to it eventually being auto-merged and updating the GitHub side membership. 
On the Contributions/forking side of things it's mainly a license check and if that passes they are free to contribute. 

We are fairly far from fully automated/self-service on the process of open sourcing new applications / code as those do need to go through a few different review processes. Though currently the open sourcing of new applications happens much less frequently than the other two above activities.   One thing that does help us with open sourcing new applications is using a repo template that helps start new projects off by providing a lot of our boiler plate and required files. Also provides a check list for the new repo owner ( https://github.com/wayfair-incubator/oss-template )

**Comment by justinabrahms** [comment]
We (eBay) use Jira. We have a custom task type called "open source request". There are inbound and outbound types. If you make an outbound type that's "I wanna release a project", we create a few subtasks via jira automation to get a security review, etc. When those are done, I'll usually go create the repo and set it up for them then make them owners.

We have janky spreadsheets to track github usernames, but one day I'm going to get SSO setup so that works for our github enterprise cloud thing.

**Reply by annania** [reply]
At my last company (Comcast), we developed a homegrown ticketing system for contributions, compliance, inner source, etc. The contribution workflow kicked off subtasks to review board members when necessary based off of OSPO initial review. Once approved, we had similar steps - one of us would have to set up the project for new projects. For contributions they would contribute from their personal account with their company email. Spreadsheets were used for github usernames and were looking to improve this. 

At (F5) we are a young OSPO, though long time involved in open source, solving this problem again. The current process has a lot of manual steps and we are looking to improve them. We are considering Jira, Smartsheets, MS PowerApps, homegrown, etc. Every company is different in how we work with varying weights to the evaluation criteria. Would love to connect with you in TODOSlack on this to hear more about your use of Jira.

**Comment by mwichmann** [comment]
I swallowed my (tongue? keyboard?) for a while on this... it's really frustrating because one of the lessons we try to learn from Open Source is it's not productive to reinvent the wheel when a community solution exists, but issue tracking seems to be one of those places (it's certainly not alone) where nothing is ever considered quite good enough.  There's always going to be a conflict between a company-wide standard, which is often a commercial product (e.g. Jira), a homegrown solution which is more tuned to a specific need, an external tool which is preferred because of integration with other things (e.g. Github - where my snarky opinion is none of their things are best-of-breed, but they win on the integration aspect), and so on. Does telling people to use something different from the company norm inhibit contribution? Does forcing people to work with a tool that doesn't quite fit inhibit usage? How do you balance? I assume there's no universal answer, but it would be cool to hear more comments.
